src/main/java/io/cdap/plugin/aws/sqs/source/SqsSourceConfig.java

```diff
@@ -0,0 +1,440 @@
+      return;
+    }
+
+    if (numberOfMessages < 1 || numberOfMessages > MAX_MESSAGES_LIMIT) {
+      collector.addFailure(String.format("Invalid number of messages '%d'.", numberOfMessages),
+        String.format("Ensure the number of messages is 1 or at most '%d'", MAX_MESSAGES_LIMIT))
+        .withConfigProperty(PROPERTY_NUMBER_OF_MESSAGES);
+    }
+  }
+
+  private void validateSchema(FailureCollector collector) {
+    if (containsMacro(PROPERTY_SCHEMA)) {
+      return;
+    }
+
+    if (Utility.isNullOrEmpty(schema)) {
+      collector.addFailure("Schema must be specified.", null)
+        .withConfigProperty(PROPERTY_SCHEMA);
+    }
+  }
+
+  private void validateFormat(FailureCollector collector) {
+    if (containsMacro(PROPERTY_SCHEMA) || containsMacro(PROPERTY_FORMAT)) {
+      return;
+    }
+
+    if (Utility.isNullOrEmpty(schema)) {
+      return;
+    }
+
+    Schema messageSchema = null;
+    try {
+      messageSchema = parseSchema();
+    } catch (Exception e) {
+      collector.addFailure(e.getMessage(), null)
+        .withConfigProperty(PROPERTY_SCHEMA);
+      collector.getOrThrowException();
+    }
+
+    // if format is empty, there must be just a single message field of type bytes or nullable
+    // types.
+    if (Utility.isNullOrEmpty(format)) {
+      List<Schema.Field> messageFields = messageSchema.getFields();
+      if (messageFields.size() > 1) {
+        List<String> fieldNames = messageFields.stream().map(o -> o.getName()).collect(Collectors.toList());
+        collector.addFailure(String.format("Without a format, the schema must contain just a single message field " +
+            "of type bytes or nullable bytes. Found %s message fields (%s).", messageFields.size(),
+          Joiner.on(',').join(fieldNames)), null)
+          .withConfigProperty(PROPERTY_SCHEMA);
+        return;
+      }
+      Schema.Field messageField = messageFields.get(0);
+      Schema messageFieldSchema = messageField.getSchema();
+      Schema.Type messageFieldType =
+        messageFieldSchema.isNullable()
+          ? messageFieldSchema.getNonNullable().getType()
+          : messageFieldSchema.getType();
+      if (messageFieldType != Schema.Type.BYTES) {
+        collector.addFailure(String.format("Without a format, the message field must be of type bytes or nullable " +
+          "bytes, but field %s is of type %s.", messageField.getName(), messageField.getSchema()), null)
+          .withConfigProperty(PROPERTY_SCHEMA);
+        return;
+      }
+    } else {
+      // otherwise, if there is a format, make sure we can instantiate it.
+      FormatSpecification formatSpec =
+        new FormatSpecification(format, messageSchema, new HashMap<String, String>());
+
+      try {
+        RecordFormats.createInitializedFormat(formatSpec);
+      } catch (Exception e) {
+        collector.addFailure(String.format("Unable to instantiate a message parser from format '%s' and message " +
```

Comment:
Can we make this error more user-friendly? This doesnt given the user clear next steps or even a clear idea what the issue is. Maybe something like :

> Current schema ([SCHEMA]) does not match the format expected for [FORMAT]: [ERROR]

Are here any common errors or corrective steps we can recommend?

Reply:
Done